compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -332,12 +332,9 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 ),
                 ExprKind::Struct(se) => {
                     let rest = match &se.rest {
-                        StructRest::Base(e) => Some(self.lower_expr(e)),
-                        StructRest::Rest(sp) => {
-                            let guar = self.dcx().emit_err(BaseExpressionDoubleDot { span: *sp });
```

Comment:
This is now gated in hir_typeck so that we can look at the struct for default fields. If there are any, then we mention enabling the feature. Otherwise we emit the same error we emitted here.